packages/client/src/runtime/core/model/applyModel.ts

```diff
@@ -93,35 +110,38 @@ export function applyModel(client: Client, dmmfModelName: string) {
       }
 
       // we handle the edge case of aggregates that need extra steps
-      if (isValidAggregateName(prop)) {
-        return applyAggregates(client, prop, action)
+      if (isValidAggregateName(key)) {
+        return applyAggregates(client, key, action)
       }
 
       return action({}) // and by default, don't override any params
     },
-    ...defaultProxyHandlers(ownKeys),
-  })
+  }
 }
 
-// the only accessible fields are the ones that are actions
 function getOwnKeys(client: Client, dmmfModelName: string) {
-  return [...Object.keys(client._baseDmmf.mappingsMap[dmmfModelName]), 'count'].filter(
-    (key) => !['model', 'plural'].includes(key),
+  const actionKeys = Object.keys(client._baseDmmf.mappingsMap[dmmfModelName]).filter(
+    (key) => key !== 'model' && key !== 'plural',
   )
-}
 
-// tells if a given `action` is valid & available for a `model`
-function isValidActionName(
-  client: Client,
-  dmmfModelName: string,
-  action: string,
-): action is Action | ClientOnlyModelAction {
-  if (isClientOnlyAction(action)) {
-    return isValidActionName(client, dmmfModelName, clientOnlyActions[action].wrappedAction)
-  }
-  return getOwnKeys(client, dmmfModelName).includes(action)
+  actionKeys.push('count')
+
+  const clientOnlyActionKeys = Object.keys(clientOnlyActions).filter((actionKey) =>
+    actionKeys.includes(clientOnlyActions[actionKey].wrappedAction),
```

Comment:
What is the reason for this change?

Reply:
It think it was a mistake during inital `*OrThrow` method implementation: they were never exposed as enumerable properties. Since now layer  does not gets called if it does not declare the specific key, error became more obvious.